### PR TITLE
Improve linting (#150)

### PR DESCRIPTION
**Pull request by aspiers**
This starts addressing #150, but we cannot hook up the new linters to Travis until the very many pylint issues are fixed.

### CODE REVIEW

File: GNUmakefile [hunk 1]
```diff
@@ -37,7 +37,10 @@ PY_FILES=$(shell $(LIST_PY_FILES))
 default: check
 
 .PHONY: check
-check: pep8 test
+check: lint test
+
+.PHONY: lint
+lint: pep8 flake8
```
Comment (Ana06) [inline GNUmakefile]: `flake8` also verifies `pep8`, right? so you are running pep8 twice. :thinking:

File: GNUmakefile [hunk 2]
```diff
@@ -49,7 +52,23 @@ pep8: $(PY_FILES)
...
+	fi
+	$(LIST_PY_FILES) | xargs flake8 --ignore=E221,E241,E251,E272,E731
+
+.PHONY: pylint
+pylint: $(PY_FILES)
+	@if ! which pylint >/dev/null 2>&1; then \
+		echo "pylint not installed!  Aborting." >&2; \
+		exit 1; \
+	fi
+	$(LIST_PY_FILES) | xargs pylint
```
Comment (Ana06) [inline GNUmakefile]: what about using `-j` option for pylint? It should be faster. :thinking:

File: GNUmakefile (hunk 1 above)
Comment (Ana06) [inline GNUmakefile]: and why not adding `pylint` here? :confused:

File: TarSCM/__init__.py [hunk 3]
```diff
@@ -31,8 +28,8 @@ def run():
 
     try:
         task_list.process_list()
-    except OptionsError as e:
-        print(e)
+    except OptionsError as exc:
```
Comment (Ana06) [inline TarSCM/__init__.py]: why this change? :confused:

File: requirements.txt [hunk 4]
```diff
@@ -1,6 +1,8 @@
+flake8
 lxml
 mock
 pep8
```
Comment (Ana06) [inline requirements.txt]: do you still want to run pep8 alone? if not this can also be removed, and some code in the GNUmakefile too.

File: TarSCM/__init__.py (hunk 3 above)
Comment (M0ses) [inline TarSCM/__init__.py]: Because pylint otherwise is complaining about invalid-name. pylint wants variable names to be at least 3 char long.

**Comment by M0ses** [comment]
could we close this PR in favor to https://github.com/openSUSE/obs-service-tar_scm/pull/171?

**Comment by M0ses** [comment]
oops, closed accidently

**Comment by aspiers** [comment]
Obsoleted by #171 and maybe others.

File: GNUmakefile (hunk 2 above)
Comment (M0ses) [inline GNUmakefile]: Thanks for your hint but it did not make any difference in my tests on my laptop (8 cores - -j 8). Took always about 5sec.
